### DISCUSSION TITLE
error: ‘::max’ has not been declared; did you mean ‘std::max’

### DISCUSSION THREAD

**Discussion by chenconghe** [ONLYOFFICE/build_tools#525] [Q&A]
i have a ubuntu22.04,and compile it , show follow error:
In file included from /home/chench/build_tools-master/tools/linux/qt-everywhere-opensource-src-5.9.9/qtbase/src/corelib/tools/qbytearray.cpp:42:
/home/chench/build_tools-master/tools/linux/qt-everywhere-opensource-src-5.9.9/qtbase/src/corelib/tools/qbytearraymatcher.h: In static member function ‘static QStaticByteArrayMatcherBase::Skiptable QStaticByteArrayMatcherBase::generate(const char*, uint)’:
/home/chench/build_tools-master/tools/linux/qt-everywhere-opensource-src-5.9.9/qtbase/src/corelib/tools/qbytearraymatcher.h:103:38: error: ‘numeric_limits’ is not a member of ‘std’
  103 |         const auto uchar_max = (std::numeric_limits<uchar>::max)();
      |                                      ^~~~~~~~~~~~~~
/home/chench/build_tools-master/tools/linux/qt-everywhere-opensource-src-5.9.9/qtbase/src/corelib/tools/qbytearraymatcher.h:103:58: error: expected primary-expression before ‘>’ token
  103 |         const auto uchar_max = (std::numeric_limits<uchar>::max)();
      |                                                          ^
/home/chench/build_tools-master/tools/linux/qt-everywhere-opensource-src-5.9.9/qtbase/src/corelib/tools/qbytearraymatcher.h:103:61: error: ‘::max’ has not been declared; did you mean ‘std::max’?
  103 |         const auto uchar_max = (std::numeric_limits<uchar>::max)();
so, wha was wrong?

**Comment by ShockwaveNN** [comment]
I think the problem in newer version of compiler in Ubuntu 22.04, could you try it with Ubuntu 16.04

This error rin qt and we cannot fix it without updating it, and we cannot update it without loosing compatibility with older systems
